### DISCUSSION TITLE
Unknown property __typename

### DISCUSSION THREAD

**Discussion by kaipiyann** [spawnia/sailor#65] [Q&A] [answered]
Hi,

First I want to say this lib is very cool, thank you Spawnia ! I use it in a project and works great to help building.

I got a problem when trying to create an input from MyInput::fromStdClass() because in the response from the API I got the field __typename for each entry. And so I don't understand why it block the creation of my ObjectLike ?

Is it a bug or is there a way to handle that ?

Thks

**Comment by kaipiyann** [comment, marked as answer]
Sorry I got my answer if works if I take the right class ! No bug of course ;-)

**Comment by kaipiyann** [comment]
Hi Spawnia,

Ok in fact now I really got a problem with __typename if I use your last version.
I explain :
You did add a check of property in ObjectLike.php
if (! property_exists($value, $name)) {
                $endpoint = static::endpoint();
                throw new InvalidDataException("{$endpoint}: Missing field {$name}.");
 }

But what happen now is that when generating the class with the shopify api, you have a property called __typename that is not always return (result from batch operation) and when I try to recreate object with fromStdClass It throw an error.

Could you advise on how I should do to handle this case ?

Thank you.

**Reply by spawnia** [reply]
Sailor always queries the field `__typename` - it is automatically added to the query string included in generated code. I would not know how to work around an API that misbehaves and does not return a queried field.
